Made development version separate to store version

soumyamahunt opened the pull request:
Made development version separate to store version, so that while both versions are installed on the same machine they can be distinguished from one another properly by their name, also command line execution for both will work at the same time. Optionally icons of the apps can also be made different for different versions.

## PR Type
What kind of change does this PR introduce?

Feature

soumyamahunt commented:
@JasonStein this takes similar approach to files-uwp to separate store version from local/development version.

JasonStein commented on src/Notepads/Package.appxmanifest:
```diff
@@ -12,14 +12,14 @@
   <mp:PhoneIdentity PhoneProductId="dde28e6c-0f89-4885-83bc-51d9580080a1" PhonePublisherId="00000000-0000-0000-0000-000000000000"/>
 
   <Properties>
-    <DisplayName>Notepads App</DisplayName>
```
I don't think you have to change this line right? Btw, I am not using Azure DevOps build to deploy to the store. I am suing local build so that I can set AppCenter secret before building the release. So in this case, I will have to make changes to switch it back from "Notepads - Dev" to "Notepads" right? Also I prefer "Notepads Dev" vs "Notepads - Dev" since it will be easier to search.

soumyamahunt replied:
> I don't think you have to change this line right? 

I have reverted it.
> Btw, I am not using Azure DevOps build to deploy to the store. I am suing local build so that I can set AppCenter secret before building the release. So in this case, I will have to make changes to switch it back from "Notepads - Dev" to "Notepads" right?

Also, you have to change the app execution alias from "Notepads-Dev" to "Notepads".
> Also I prefer "Notepads Dev" vs "Notepads - Dev" since it will be easier to search.

Made this change.

soumyamahunt commented:
@JasonStein Also, it would be nicer if someone makes a new icon for development version. Just adding "Dev" label to the current icon would suffice.

JasonStein commented:
> @JasonStein Also, it would be nicer if someone makes a new icon for development version. Just adding "Dev" label to the current icon would suffice.

@mah-qurashy Can you help make one when you have time? This is not high priority.

JasonStein commented on azure-pipelines.yml:
```diff
@@ -13,6 +13,18 @@ variables:
+    script: |
+      [xml]$manifest = Get-Content $(Build.SourcesDirectory)\src\Notepads\Package.appxmanifest
+      $manifest.Package.Identity.Name="Notepads"
+      $manifest.Package.Properties.DisplayName="Notepads App"
```
Do we need this line?

JasonStein replied:
Notepads App ?

soumyamahunt replied:
We don't need it anymore, since we aren't changing it. I have removed it.

mah-qurashy commented:
> @mah-qurashy Can you help make one when you have time? This is not high priority.

yes i could do it, do you have any ideas about how the icon should look?

soumyamahunt commented:
> yes i could do it, do you have any ideas about how the icon should look?

May be adding a simple label like this:
![Files UWP Beta Icon](https://user-images.githubusercontent.com/51203900/89125312-39cf7b00-d4fb-11ea-9579-3445a56cf581.png)

JasonStein commented:
> > @mah-qurashy Can you help make one when you have time? This is not high priority.
> 
> yes i could do it, do you have any ideas about how the icon should look?

Add a label like above but name it as "Dev"

mah-qurashy commented:
let me know what do you think about these
![image](https://user-images.githubusercontent.com/20822445/90081553-f1d80180-dd0d-11ea-8c14-3f8f36c5efc2.png)

JasonStein commented:
> let me know what do you think about these
> ![image](https://user-images.githubusercontent.com/20822445/90081553-f1d80180-dd0d-11ea-8c14-3f8f36c5efc2.png)

LGTM.

soumyamahunt commented:
> LGTM.

Great then, @mah-qurashy can you submit a pr targeting this branch or share the generated assets files.

mah-qurashy commented:
> Great then, @mah-qurashy can you submit a pr targeting this branch or share the generated assets files.

i will start working on all the sizes, it would take some time, should i submit pr to your [/dev](https://github.com/soumyamahunt/Notepads/tree/dev) branch when done?

soumyamahunt commented:
> i will start working on all the sizes, it would take some time, should i submit pr to your [/dev](https://github.com/soumyamahunt/Notepads/tree/dev) branch when done?

Yes.

soumyamahunt commented:
@JasonStein thanks to @mah-qurashy the development version of assets files has been added. Right now the project configuration contains 3 values: Debug, Release, Production.
![devenv_C78NfBjPSd](https://user-images.githubusercontent.com/51203900/100075730-5a369300-2e66-11eb-9fb1-d4604d677ade.png)
Choosing either "Debug" or "Release" will produce development version while choosing "Production" will produce the store version. I have modified project file and added a powershell script to handle display names and assets linking, so while generating packages for store the only thing you have to do is switch to "Production" build configuration.

JasonStein commented on src/Notepads/Notepads.csproj:
```diff
@@ -860,6 +1016,10 @@
   </PropertyGroup>
   <Import Project="$(MSBuildExtensionsPath)\Microsoft\WindowsXaml\v$(VisualStudioVersion)\Microsoft.Windows.UI.Xaml.CSharp.targets" />
+  <PropertyGroup Condition="'$(Configuration)' == 'Production'">
+    <PreBuildEvent>powershell -ExecutionPolicy Bypass $(ProjectDir)build.ps1 -project_dir $(ProjectDir) -target PreBuild</PreBuildEvent>
```
Why it has both Pre and Post-build under Production section?

soumyamahunt replied:
Pre-build event modifies the manifest to be used for Store release, post-build event restores the manifest to default.

JasonStein commented:
Ready to merge?

soumyamahunt commented:
> Ready to merge?

Yes.

soumyamahunt commented:
@JasonStein I have simplified the build scripts, this PR is now ready to merge.
